lib/mailosaur.rb

```diff
@@ -38,37 +38,45 @@ module Models
   end
 
   class MailosaurClient
+    #
+    # Creates initializes a new instance of the MailosaurClient class.
+    # @param api_key [String] your Mailosaur API key.
+    # @param base_url [String] the base URI of the service.
+    #
+    def initialize(api_key, base_url = 'https://mailosaur.com/')
+      @api_key = api_key
+      @base_url = base_url
+    end
+
     # @return [Analysis] analysis
-    attr_reader :analysis
+    def analysis
+      @analysis ||= Analysis.new(connection)
+    end
 
     # @return [Files] files
-    attr_reader :files
+    def files
+      @files ||= Files.new(connection)
+    end
 
     # @return [Messages] messages
-    attr_reader :messages
+    def messages
+      @messages ||= Messages.new(connection)
+    end
 
     # @return [Servers] servers
-    attr_reader :servers
+    def servers
+      @servers ||= Servers.new(connection)
+    end
 
-    #
-    # Creates initializes a new instance of the MailosaurClient class.
-    # @param api_key [String] your Mailosaur API key.
-    # @param base_url [String] the base URI of the service.
-    #
-    def initialize(api_key, base_url)
-      conn = Faraday.new(base_url || 'https://mailosaur.com/', {
+    private
+
+    def connection
+      Faraday.new(@base_url, {
         headers: {
           content_type: 'application/json; charset=utf-8',
-          user_agent: 'mailosaur-ruby/' + Mailosaur::VERSION
+          user_agent: 'mailosaur-ruby/5.0.0'
```

Comment:
Just need this line reverted please 👍

Reply:
:man_shrugging: I've no idea why it's there, but sure thing